lib/screenplay_web/auth_manager/ensure_pa_message_admin.ex

```diff
@@ -0,0 +1,22 @@
+defmodule ScreenplayWeb.EnsurePaMessageAdmin do
+  @moduledoc """
+  Verify that the user has permission to access the PA Message Creation feature.
+  """
+
+  import Plug.Conn, only: [halt: 1]
+  import Phoenix.Controller, only: [redirect: 2]
+
+  def init(options), do: options
+
+  def call(conn, _opts) do
+    with claims <- Guardian.Plug.current_claims(conn),
+         true <- :pa_message_admin in ScreenplayWeb.AuthManager.claims_access_level(claims) do
```

Comment:
We seem to be doing this same logic in quite a lot of places: here, the other `ensure` plug, twice in `metadata`... can we do it once in a common plug (maybe `metadata`) and put the "finished" list of roles in the conn? The `ensure` plugs would still need to exist, but could just check if e.g. `:pa_message_admin in roles`.

Reply:
Good call. Will move some code around to make this work.